sdk/eventhub/event-hubs/test/internal/bufferedProducerUnitTests.spec.ts

```diff
@@ -1,92 +1,56 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import { EventHubBufferedProducerClient } from "../../src";
-import { EnvVarKeys, getEnvVars } from "../public/utils/testUtils";
-import { createStubInstance, SinonStubbedInstance, stub } from "sinon";
-import chai from "chai";
-import { createMockServer } from "../public/utils/mockService";
-import { testWithServiceTypes } from "../public/utils/testWithServiceTypes";
-import { delay } from "@azure/core-amqp";
-import { BatchingPartitionChannel } from "../../src/batchingPartitionChannel";
-import { PartitionAssigner } from "../../src/impl/partitionAssigner";
+import type { EventHubBufferedProducerClient } from "../../src/index.js";
+import { createBufferedProducer } from "../utils/clients.js";
+import { describe, it, afterEach, vi, beforeEach, beforeAll, afterAll } from "vitest";
+import { assert, expect } from "../utils/chai.js";
 
-const should = chai.should();
-const assert: typeof chai.assert = chai.assert;
+describe("EventHubBufferedProducerClient unit tests", function () {
+  let client: EventHubBufferedProducerClient;
 
-testWithServiceTypes((serviceVersion) => {
-  const env = getEnvVars();
-  if (serviceVersion === "mock") {
-    let service: ReturnType<typeof createMockServer>;
-    before("Starting mock service", () => {
-      service = createMockServer();
-      return service.start();
-    });
+  beforeAll(async function () {
+    vi.useFakeTimers();
+  });
 
-    after("Stopping mock service", () => {
-      return service?.stop();
-    });
-  }
+  afterAll(async function () {
+    vi.restoreAllMocks();
+    vi.useRealTimers();
+  });
 
-  describe("EventHubBufferedProducerClient unit tests", () => {
-    const service = {
-      connectionString: env[EnvVarKeys.EVENTHUB_CONNECTION_STRING],
-      path: env[EnvVarKeys.EVENTHUB_NAME]!,
-    };
-    let client: EventHubBufferedProducerClient | undefined;
+  beforeEach(async function () {
+    client = createBufferedProducer().producer;
+  });
 
-    before(() => {
-      should.exist(
-        env[EnvVarKeys.EVENTHUB_CONNECTION_STRING],
-        "define EVENTHUB_CONNECTION_STRING in your environment before running integration tests.",
-      );
-      should.exist(
-        env[EnvVarKeys.EVENTHUB_NAME],
-        "define EVENTHUB_NAME in your environment before running integration tests.",
-      );
-    });
+  afterEach(async function () {
```

Comment:
do we need to set client to undefined like previous code?

Reply:
I guess no, the client is being constructed before each test and I don't see a reason to set it to undefined first before constructing it again.